Docs: add restart-policies to 'Run reference'

thaJeztah opened the pull request:
This moves some information on restart-policies from the "command line" page to "run reference".

Also fixes some minor typos and adds a "NOTE" about --rm and --restart not allowed to be combined.

I decided to move the detailed information from the command-line page to the run-reference, because the command-line page is already quite lengthy. The information on there was (imo) good enough to be included on the "run reference" page.

Also removes inline CSS styles from tables, which will be styled by the stylesheet.

depends on: https://github.com/docker/docs-base/pull/1

resolves #11069

thaJeztah committed:
Docs: add restart-policies to 'Run reference'

This moves some information on restart-policies from
the "command line" page to "run reference".

Also fixes some minor typos and adds a "NOTE"
about --rm and --restart not allowed to be combined.

Also removes inline CSS styles from tables,
which will be styled by the stylesheet, and fixes
some minor MarkDown errors (`<` -> &lt;)

depends on https://github.com/docker/docs-base/pull/1

resolves #11069

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

moxiegirl commented:
What is doing the restarting in this case? I assume it is the `docker` daemon. If that is true, is there a reason we wouldn't mention that this is the case?

I would put the flags in a table.  The page is hard enough to read as it is and reference material in a table is easy to spot.

Finally, regarding **always** it doesn't have the optional max-retries to set.  So, just how long will a retry on failure go under **--always**?   

I would move the material you have in the note after the table and before the example. This isn't really a note it is important information that applies to any restart setting. 

Looks like a good move to consolidate all this restart material in one place. :-D

----
#### Restart Policies

Use Docker's `--restart` to specify a container's _restart policy_.  A restart policy controls whether the `docker` daemon restarts a container after exit. Docker supports the following restart policies:

<style type="text/css">

</style>
<table class="tg">
  <tr>
    <td class="tg-031e"><strong>no</strong></th>
    <td class="tg-031e">Do not restart. This is the default.</th>
  </tr>
  <tr>
    <td class="tg-031e"><strong>on-failure</strong>[<strong>:</strong><emphasis>max-retries</emphasis>]</td>
    <td class="tg-031e">Restart only if the container exists with a non-zero exit status. Optionally, limit the number of restart retries the `docker` daemon attempts.</td>
  </tr>
  <tr>
    <td class="tg-031e"><strong>always</strong></td>
    <td class="tg-031e">Always restart the container regardless of the exit status. When you specify always, the `docker` daemon repeatedly tries retries until XXXX.</td>
  </tr>
</table>

You cannot set any restart policy in combination with ["clean up (--rm)"](#clean-up-rm). Setting both `--restart` and `--rm` results in an  error.

The following example runs a `redis` container with a restart policy of **-always**

    $ sudo docker run --restart=always redis

If the container exits, Docker restart its for you.

thaJeztah commented:
@moxiegirl thanks for reviewing. I will do some rewriting with your suggestions added. (The current PR is only a very minimal modification of the existing text)

I like the table idea (also for other parts); does the docs markdown processor support tables? I know the one on the GitHub registry doesn't, but I'm not sure that's the same one.

moxiegirl commented:
@thaJeztah  When you run "make docs" you can check the table output.  I've been using them and they come out fine in the test.  I use this http://www.tablesgenerator.com/html_tables.

thaJeztah commented:
@moxiegirl ah, embedded html `<table>`. That could work, not sure native "MarkDown" tables are supported, probably that was what the issue was about then (http://www.tablesgenerator.com/markdown_tables), but I can check if it that works

thaJeztah commented:
Updated with suggestions by @moxiegirl.

NOTE: this currently depends on https://github.com/docker/docs-base/pull/1 to be merged first, because of stylesheet changes. If we want this in before that is merged, I can temporarily add a `<style>` tag to these documents.

moxiegirl commented:
@thaJeztah Slight issue in run.md with the bolds. 
![image](https://cloud.githubusercontent.com/assets/1347209/6478293/b9f505dc-c1e4-11e4-8741-d97441fcc1d3.png)
Quick fix to that and LGTM

thaJeztah commented:
@VynceMontgomery Good questions, I'll look up if the interval is reset on a successful start. I think there was word on a restart count, but I'll look that up as well

thaJeztah commented:
@MoxieGirl ah, yes, that's a difference in MarkDown handling between GitHub and mkdocs. I corrected some, but apparently missed those, thx

thaJeztah commented:
@moxiegirl @VynceMontgomery I made some changes;

- Added a description that the 'delay' is reset after a successful (re)start
- Added an example to see the number of restarts and the last time the container was (re)started
- Removed a leftover `class="code"` that was used for testing stylesheet changed :ski

thaJeztah commented:
Sorry, mouse-slip wrong button

moxiegirl commented:
@SvenDowideit @fredlf @jamtur01 @spf13 One more LGTM needed...

jamtur01 commented:
LGTM
